traits/src/float.rs

```diff
@@ -950,264 +1094,312 @@ macro_rules! float_impl {
             fn round(self) -> Self {
                 <$T>::round(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn trunc(self) -> Self {
                 <$T>::trunc(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn fract(self) -> Self {
                 <$T>::fract(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn abs(self) -> Self {
                 <$T>::abs(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn signum(self) -> Self {
                 <$T>::signum(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn is_sign_positive(self) -> bool {
                 <$T>::is_sign_positive(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn is_sign_negative(self) -> bool {
                 <$T>::is_sign_negative(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn mul_add(self, a: Self, b: Self) -> Self {
                 <$T>::mul_add(self, a, b)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn recip(self) -> Self {
                 <$T>::recip(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn powi(self, n: i32) -> Self {
                 <$T>::powi(self, n)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn powf(self, n: Self) -> Self {
                 <$T>::powf(self, n)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn sqrt(self) -> Self {
                 <$T>::sqrt(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn exp(self) -> Self {
                 <$T>::exp(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn exp2(self) -> Self {
                 <$T>::exp2(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn ln(self) -> Self {
                 <$T>::ln(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn log(self, base: Self) -> Self {
                 <$T>::log(self, base)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn log2(self) -> Self {
                 <$T>::log2(self)
             }
 
+            #[cfg(feature = "std")]
             #[inline]
             fn log10(self) -> Self {
                 <$T>::log10(self)
             }
 
             #[inline]
             fn to_degrees(self) -> Self {
-                // NB: `f32` didn't stabilize this until 1.7
-                // <$T>::to_degrees(self)
-                self * (180. / ::std::$T::consts::PI)
+                <$T>::to_degrees(self)
```

Comment:
Oh, this doesn't work for no_std, because the inherent methods are in std.  So it recurses:

```rust
warning: function cannot return without recurring
    --> src\float.rs:1272:13
     |
1272 |               fn to_degrees(self) -> Self {
     |  _____________^ starting here...
1273 | |                 <$T>::to_degrees(self)
1274 | |             }
     | |_____________^ ...ending here
...
1482 |   float_impl!(f32 integer_decode_f32 classify_f32);
     |   ------------------------------------------------- in this macro invocation
     |
     = note: #[warn(unconditional_recursion)] on by default
```

I guess we have to roll our own, but it's not a big deal since these are trivial.

Can you add `#![deny(unconditional_recursion)]` to the crate?  This seems like a problem we're vulnerable to in general, with so many traits that are supposed to forward calls...